src/EditorFeatures/Core/Implementation/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -159,6 +188,316 @@ private string GetRootPathWithGuid_NoLock()
             return new MetadataAsSourceFile(fileInfo.TemporaryFilePath, navigateLocation, documentName, documentTooltip);
```

Comment:
it's interesting that you coudl start with a VB document, but end up creating a C# MAS file.  Probably worth adding a test to make sure that things work properly in that cases.  i.e. that we end up making a .cs file in a C# project and not a VB project.